upgrades/upgrade_testing_framework/robot_lib/OpenSearchRESTActions.py

```diff
@@ -1,34 +1,44 @@
 from pathlib import Path
 import json
 from typing import Any
-
-from opensearchpy import OpenSearch
+import cluster_migration_core.core.versions_engine as ev
+import cluster_migration_core.clients as clients
 
 DEFAULT_TEMP_STORAGE = "/tmp/utf/robot_data.json"
 
 
-# Change this library to use REST requests instead of Python client
-class OpenSearchClientLibrary(object):
+class CouldNotRetreiveAttributeFromRESTResponse(Exception):
+    def __init__(self, key: str, response: dict):
+        super().__init__(f"Could not find key '{key}' in the RESTResponse json: {response}")
+
+
+class OpenSearchRESTActions(object):
 
     def __init__(self, host="localhost", port=9200, temp_storage_location=DEFAULT_TEMP_STORAGE):
-        os_client = OpenSearch(hosts=[{'host': host, 'port': port}])
-        self._os_client = os_client
+        #This is temporary, this will be provided by the executor
+        engine_version = ev.EngineVersion(ev.ENGINE_OPENSEARCH, 1, 3, 6)
+        rest_client = clients.get_rest_client(engine_version)
+        self._rest_client = rest_client
         self._temp_storage_location = Path(temp_storage_location)
+        self._port = port
 
     def create_index(self, index_name: str):
-        self._os_client.indices.create(index_name)
-
-    def create_document_in_index(self, index_name: str, document: str):
-        self._os_client.index(index=index_name, body=document)
+        self._rest_client.create_an_index(port=self._port, index=index_name)
 
-    def execute_query_on_index(self, index_name: str, query: str) -> str:
-        return self._os_client.search(index=index_name, body=query)
+    def create_document(self, index_name: str, document: dict):
+        self._rest_client.post_doc_to_index(port=self._port, index=index_name, doc=document)
 
     def count_documents_in_index(self, index_name: str) -> int:
-        return self._os_client.count(index=index_name).get('count')
+        response = self._rest_client.count_doc_in_index(port=self._port, index=index_name).response_json
+        count_key = "count"
+        try:
+            count = response[count_key]
+            return count
+        except CouldNotRetreiveAttributeFromRESTResponse(count_key, response):
+            print('Please enter a valid attribute from RESTResponse')
```

Comment:
Hmmm, not sure I understand your intent here.  Can you clarify?  I actually didn't know you even could supply an instantiated object to an `except` clause like this.  Also, curious why we're using a print statement instead of Python's logging facilities.

Reply:
Ah, thanks for pointing this out, I had written the whole try catch block in a different way which ended up being reverted when I was undoing some of the code I wrote while testing a few other things.
But the idea is to throw an exception when `count_key != "count"` and provide the value of count_key and the response gotten from the REST client.
As for the print statement, it's also part of the things that were undone, I do realize though that it's considered bad practice and it's better to use python logging at least for categorizing the messages, So thanks for pointing that out as well :)